CoreScriptsRoot/CoreScripts/Topbar.lua

```diff
@@ -70,6 +74,8 @@ while not Player do
 	Player = PlayersService.LocalPlayer
 end
 
+local canChat = true
```

Comment:
How can you guarantee that canChat has been set before the setting is read? It doesn't matter in this case because you are removing the top bar button when canChat is set but the canChat variable seems potentially misleading.